sdk/src/main/java/module-info.java

```diff
@@ -0,0 +1,22 @@
+module com.hedera.hashgraph.sdk {
+    requires transitive com.google.common;
+    requires transitive com.google.protobuf;
+    requires transitive headlong;
+    requires transitive io.grpc.stub;
+    requires transitive io.grpc;
+    requires transitive org.bouncycastle.provider;
+    requires transitive org.slf4j;
```

Comment:
Why those are transitive?

Reply:
The `checkAllModuleInfo` dependency scope check tells us if a type of these dependencies is used in public API. Then the dependency should be transitive.

You can run `./gradlew :sdk:reason --id com.google.guava:guava` (or another `--id`) for details and then search the code base for the places. Maybe some things are unintentional.

Reply:
@jjohannes I think to leave same transitive dependencies as we have now and add new transitive dependencies as a follow up item later along with communicating it to the community as it might potentially break apps.

Reply:
That makes sense.

I changed this to only have `com.google.protobuf` (com.google.protobuf:protobuf-javalite) as _transitive_ (api) dependency as it currently is on `main`.

For the analysis, I added the corresponding configuration in [build.gradle.kts](https://github.com/hashgraph/hedera-sdk-java/blob/gradle-updates/build.gradle.kts#L7-L16) you can then decide yourself what to do for each case (or just leave things as they are).

Reply:
Thanks @jjohannes 

I think we should only have protobuf dependency as transitive now without exposing others like in [build.gradle.kts](https://github.com/hashgraph/hedera-sdk-java/blob/gradle-updates/build.gradle.kts#L7-L16) 
So its 1:1 like in `main`

Reply:
Yes that is the case after my changes. Only [`com.google.protobuf` is transitive](https://github.com/hashgraph/hedera-sdk-java/blob/gradle-updates/sdk/src/main/java/module-info.java#L2).

The dependency analysis plugin configuration in [build.gradle.kts](https://github.com/hashgraph/hedera-sdk-java/blob/gradle-updates/build.gradle.kts#L7-L16) are required to tell the dependency analysis to ignore that certain classes expose types from dependencies in their API although we do not declare those dependencies as transitive.